session/src/main/java/org/apache/iotdb/session/Session.java

```diff
@@ -1096,6 +1118,10 @@ private void insertRecordsWithLeaderCache(
         handleRedirection(entry.getKey(), e.getEndPoint());
       } catch (StatementExecutionException e) {
         errMsgBuilder.append(e.getMessage());
+      } catch (IoTDBConnectionException e) {
```

Comment:
Maybe we can optimize insertRecords interface to perfrom one request per sessionConnection rather than per deviceID, just like insertTablets.

Reply:
Ok I'll optimize later in this PR.